lib/crypto/packaged.py

```diff
@@ -32,95 +31,71 @@ def get_endpoint(file_obj):
+    log.debug('Calling signing service: {0}'.format(endpoint))
+    with statsd.timer('services.sign.addon'):
+        response = requests.post(endpoint,
+                                 timeout=settings.SIGNING_SERVER_TIMEOUT,
+                                 data={'addon_id': addon_id},
+                                 files={'file': ('mozilla.sf',
+                                                 str(jar.signatures))})
     if response.status_code != 200:
-        msg = 'Posting to add-on signing failed: %s' % response.reason
+        msg = 'Posting to add-on signing failed: {0}'.format(response.reason)
         log.error(msg)
         raise SigningError(msg)
 
     pkcs7 = b64decode(json.loads(response.content)['mozilla.rsa'])
-    try:
-        cert_serial_num = get_signature_serial_number(pkcs7)
-        jar.make_signed(pkcs7, sigpath='mozilla')
-    except:
-        msg = 'Addon signing failed'
-        log.error(msg, exc_info=True)
-        raise SigningError(msg)
+    cert_serial_num = get_signature_serial_number(pkcs7)
+    jar.make_signed(pkcs7, sigpath='mozilla')
     shutil.move(temp_filename, file_obj.file_path)
     return cert_serial_num
 
 
 def sign_file(file_obj):
-    try:
-        cert_serial_num = call_signing(file_obj)  # Sign file.
-        if cert_serial_num:
-            # Save the certificate serial number for revocation if needed, and
-            # re-hash the file now that it's been signed.
-            file_obj.update(cert_serial_num=cert_serial_num,
-                            hash=file_obj.generate_hash())
-    except SigningError:
-        log.info('Signing failed for file %s.' % file_obj.pk)
-        raise
-    log.info('Signing complete for file %s.' % file_obj.pk)
-
-
-@task
-def sign(version):
-    if not version.all_files:
-        log.error(
-            'Attempt to sign version %s with no files.' % version.pk)
-        raise SigningError('No file')
-
-    log.info('Signing version: %s' % version.pk)
-
-    for file_obj in [x for x in version.all_files]:
-        with statsd.timer('services.sign.addon'):
-            sign_file(file_obj)
+    """Sign a File.
+
+    If the file is a hotfix, or isn't reviewed yet, or there's no endpoint
+    (signing not enabled), or there was an error while signing, log and return
+    nothing.
+
+    Otherwise return the signed file.
+    """
+    # Don't sign hotfixes.
+    if file_obj.version.addon.guid in settings.HOTFIX_ADDON_GUIDS:
+        log.info('Not signing file {0}: addon is a hotfix'.format(file_obj.pk))
+        return
+
+    # We only sign files that have been reviewed.
+    if file_obj.status not in [amo.STATUS_LITE, amo.STATUS_PUBLIC]:
+        log.info("Not signing file {0}: it isn't reviewed".format(file_obj.pk))
+        return
+
+    endpoint = get_endpoint(file_obj)
+    if not endpoint:  # Signing not enabled.
+        log.info('Not signing file {0}: no active endpoint'.format(
+            file_obj.pk))
+        return
```

Comment:
It seems like this should go first to me. If it isn't configured none of the others matter.